gno.land/pkg/gnoweb/gnoweb.go

```diff
@@ -335,6 +336,15 @@ func handleRealmRender(logger *slog.Logger, app gotuna.App, cfg *Config, w http.
 			return
 		}
 	}
+
+	dirdata := []byte(rlmpath)
+	dirres, err := makeRequest(logger, cfg, qFileStr, dirdata)
+	if err != nil {
+		writeError(logger, w, err)
+		return
+	}
+	hasReadme := bytes.Contains(dirres.Data, []byte("README.md\n"))
```

Comment:
If `README.md` file is the last element in the list (or the only one), it may not work, as the last element does not have a new line at the end.

Reply:
Oh, I see. That was your reasoning. It might be correct in the current state, but there is a high probability that Gnoweb will show a realm that does not necessarily have a `gno.mod`. For example, I'm working on implementing lazy loading in `gnodev`, and I will allow realms that don't have `gno.mod` to be published. There is also an ongoing discussion to decide whether `gno.mod` has any utility on the chain (since we don't have real versioning _yet_).

In any case, I feel that not depending on any specific ordering is better to avoid some weird behaviours later.

> However, if you still think that we should check this case, would this be a good way to do so ? Or am I overthinking it ?

Don't waste too much time on this, I would simply suggest adding a trailing newline with `append`:

```suggestion
	hasReadme := bytes.Contains(append(dirres.Data, '\n'), []byte("README.md\n"))
```

Reply:
Thanks for the help ! That makes sense !